src/subhalo_tracking.cpp

```diff
@@ -528,14 +528,30 @@ void SubhaloSnapshot_t::FeedCentrals(HaloSnapshot_t &halo_snap)
       assert(central.Particles.size());
       central.Particles.swap(Host.Particles); // reuse the halo particles
       central.Nbound = central.Particles.size();
+      bool tracerIndexSet = false;
+      // Use the most bound tracer from the previous snapshot that remains
+      // in the FOF group as the tracer. We need this information for the masking.
+      for (HBTInt i = central.GetTracerIndex(); i < Host.Particles.size(); i++)
```

Comment:
I don't think TracerIndex is valid at this point because of this part a few lines earlier which replaces the particle array with the FoF particles:
```
central.Particles.swap(Host.Particles); // reuse the halo particles
```
But we can call GetTracerIndex earlier and store the result.

Reply:
I'm indexing `Host.Particles` below to find the most bound particle ID, so I think this should be ok.

Reply:
Maybe it's still worth moving the GetTracerIndex call earlier so that we can check that the tracer index is correct before we use it?

Reply:
Yep I agree, I've moved it